src/generic.rs

```diff
@@ -999,14 +999,14 @@ where
     )?;
 
     let future_tx1 = PyObject::from(py_fut.clone());
-    let future_tx2 = future_tx1.clone();
+    let future_tx2 = future_tx1.clone_ref(py);
 
     R::spawn_local(async move {
-        let locals2 = locals.clone();
+        let locals2 = Python::with_gil(|py| locals.clone_ref(py));
 
         if let Err(e) = R::spawn_local(async move {
             let result = R::scope_local(
-                locals2.clone(),
+                Python::with_gil(|py| locals2.clone_ref(py)),
```

Comment:
I think these patterns are all correct for now. It would be interesting to explore further what we can do to reduce calls to `Python::with_gil`, and the correct way to release the GIL across yield points too.

I think they are follow up areas of research, which probably can yield new APIs both here and upstream in PyO3.

(I guess there's potentially a lot of interesting performance benchmarking to do?)